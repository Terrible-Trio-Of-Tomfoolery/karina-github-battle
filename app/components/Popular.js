import React, {Component} from 'react';

export default class Popular extends Component {
    render() {
        const languages = ['All', 'JavaScript', 'Ruby', 'Java', 'CSS', 'Python'];

        return (
            <ul className='flex-centre'>
                {languages.map(language => (
                    <li key={language}>
                        <button className='btn-clear nav-link'>
                            {language}
                        </button>
                    </li>
                ))
                }
            </ul>
        )
    }
}
