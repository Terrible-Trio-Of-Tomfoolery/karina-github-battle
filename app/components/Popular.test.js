import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

import Popular from "./Popular";

describe('Popular', () => {
    let wrapper;
    beforeAll(() => {
        wrapper = shallow(<Popular/>);
    });
    it('does not crash on render', () => {
        expect(wrapper.find('ul')).not.toBeNull();
    });
    it('renders correctly', () => {
        expect(renderer.create(<Popular/>).toJSON()).toMatchSnapshot();
    });
});
