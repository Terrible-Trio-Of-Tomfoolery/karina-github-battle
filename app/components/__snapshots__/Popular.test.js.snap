exports[`Popular renders correctly 1`] = `
<ul
  className="flex-centre"
>
  <li>
    <button
      className="btn-clear nav-link"
    >
      All
    </button>
  </li>
  <li>
    <button
      className="btn-clear nav-link"
    >
      JavaScript
    </button>
  </li>
  <li>
    <button
      className="btn-clear nav-link"
    >
      Ruby
    </button>
  </li>
  <li>
    <button
      className="btn-clear nav-link"
    >
      Java
    </button>
  </li>
  <li>
    <button
      className="btn-clear nav-link"
    >
      CSS
    </button>
  </li>
  <li>
    <button
      className="btn-clear nav-link"
    >
      Python
    </button>
  </li>
</ul>
`;
